integrations/adobe-analytics/lib/index.js

```diff
@@ -139,10 +146,83 @@ AdobeAnalytics.prototype.initialize = function() {
   window._segHBPlayheads = {};
 
+  // Load the more compact Chromecast SDK only if the customer has it enabled in settings
+  if (options.chromecastMode) {
+
+    window.ADBmobile = {};
+    window.ADBmobile.analytics = {};
+    window.ADBMobileConfig = {
+      "marketingCloud": {
+        "org": `${options.marketingCloudOrgId}`
+      },
+      "target": {
+        "clientCode": "",
+        "timeout": 5
+      },
+      "audienceManager": {
+        "server": `${options.trackingServerUrl}`
+      },
+      "analytics": {
+        "rsids": `${options.reportSuiteId}`,
+        "server": `${options.trackingServerUrl}`,
+        "ssl": false,
+        "offlineEnabled": false,
+        "charset": "UTF-8",
+        "lifecycleTimeout": 300,
+        "privacyDefault": "optedin",
+        "batchLimit": 0,
+        "timezone": "MDT",
+        "timezoneOffset": -360,
+        "referrerTimeout": 0,
+        "poi": []
+      },
+      "mediaHeartbeat": {
+        "server": `${options.heartbeatTrackingServerUrl}`,
+        "publisher": "B3CB46FC57C6C8F77F000101@AdobeOrg",
+        "channel": "test-channel-chromecast",
+        "ssl": true,
+        "ovp": "chromecast-player",
+        "sdkVersion": "chromecast-sdk",
+        "playerName": "Chromecast"
+      }
+    };
+
+    self.mediaHeartbeats = {};
+    self.adBreakCounts = {};
+    self.qosData = {};
+    self.playhead = 0;
+    self.adBreakInProgress = false;
+
+    self.heartbeatEventMap = {
+      // Segment spec'd event: Heartbeat function
+      'video playback started': chromecastInit,
+      'video playback paused': chromecastVideoPaused,
+      'video playback resumed': chromecastVideoStart, // Treated as a 'play' as well.
+      'video playback buffer started': chromecastBufferStarted,
+      'video playback buffer completed': chromecastBufferCompleted,
+      'video playback seek started': chromecastSeekStarted,
+      'video playback seek completed': chromecastSeekCompleted,
+      'video playback completed': chromecastSessionEnd,
+      'video playback interrupted': chromecastVideoPaused,
+      'video quality updated': chromecastQualityUpdated,
+      'video content started': chromecastHeartbeatVideoStart,
+      'video content playing': chromecastUpdatePlayhead,
+      'video content completed': chromecastVideoComplete,
+      'video ad started': chromecastAdStarted,
+      'video ad skipped': chromecastAdSkipped,
+      'video ad completed': chromecastAdCompleted,
+      'video playback exited': chromecastPlaybackExited
+    };
+
+
+    this.load('chromecast', function () {
+      self.ready();
+    });
+  }
   // Load the larger Heartbeat script only if the customer has it enabled in settings.
   // This file is considerably bigger, so this check is necessary.
-  if (options.heartbeatTrackingServerUrl) {
-    this.load('heartbeat', function() {
+  if (!options.chromecastMode && options.heartbeatTrackingServerUrl) {
```

Comment:
Prefer if we did not add `!options.chromecastMode` to this conditional and next.

Since this `AdobeAnalytics.prototype.initialize` is quite large I propose we break out chromecast initialization separately.

```js
AdobeAnalytics.prototype.initialize = function () {
// initial setup
if (options.chromecastMode) {
  this.initializeChromeCastMode()
} else if (options.heartbeatTrackingServerUrl) {
  // existing heartbeat init
} else {
  // existing default init
}
```

Reply:
@kdaswani  - I'll leave this one hardcoded and unresolved and let you and @mericsson discuss/update

Reply:
will update this to swap the conditional!